src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/posthearingreview/PostHearingReviewAboutToSubmitHandler.java

```diff
@@ -37,15 +47,35 @@ public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType,
                                                           String userAuthorisation) {
         SscsCaseData caseData = callback.getCaseDetails().getCaseData();
 
-        PreSubmitCallbackResponse<SscsCaseData> response = new PreSubmitCallbackResponse<>(caseData);
-
         String caseId = caseData.getCcdCaseId();
 
         PostHearingReviewType typeSelected = caseData.getPostHearing().getReviewType();
         log.info("Review Post Hearing App: handling action {} for case {}", typeSelected,  caseId);
 
+        PreSubmitCallbackResponse<SscsCaseData> response = new PreSubmitCallbackResponse<>(caseData);
+
+        if (PostHearingReviewType.SET_ASIDE.equals(typeSelected)) {
+            addDocument(caseData, SET_ASIDE_APPLICATION);
+        }
+
         SscsUtil.clearDocumentTransientFields(caseData);
 
         return response;
     }
+
+    private void addDocument(SscsCaseData caseData, DocumentType documentType) {
```

Comment:
Put this as a static helper method in SscsUtil and use here and in ActionPostponementRequestAboutToSubmitHandler

Reply:
done